bitcoincore/src/main/kotlin/io/horizontalsystems/bitcoincore/models/Checkpoint.kt

```diff
@@ -7,20 +7,28 @@ import java.io.BufferedReader
 import java.io.InputStreamReader
 import java.io.Reader
 
-class Checkpoint(fileName: String) {
+class Checkpoint(fileName: String, providedBlock: Block? = null) {
     val block: Block
     val additionalBlocks: List<Block>
 
     init {
-        val stream = javaClass.classLoader?.getResourceAsStream(fileName)
-        val inputStreamReader: Reader = InputStreamReader(stream)
-        val reader = BufferedReader(inputStreamReader)
-        val checkpoints = reader.readLines()
+        if (providedBlock != null) {
+            val blocks = listOf(providedBlock)
 
-        val blocks = checkpoints.map { readBlock(it) }
+            block = blocks.first()
+            additionalBlocks = blocks.drop(1)
```

Comment:
Is this not the same as doing this?
```suggestion
            block = providedBlock
            additionalBlocks = List<Block>()
```
I'm not sure if my syntax is correct; I just want to make sure I understand the logic.

Reply:
I believe so.